### ISSUE TITLE
digistuff channel copier is very limited

### ISSUE THREAD

**Issue by SwissalpS** [pandorabox-io/pandorabox.io#550]
The current code states that [other mods should add some field to be compatible.](https://github.com/minetest-mirrors/digistuff/blob/c4b40bcfd5c7116a48af87e3db07a9c29757fed9/channelcopier.lua#L62)
I think that is not a nice way to go about it.

I have [forked it and added some more nodes](https://github.com/SwissalpS/digistuff/tree/betterCompat). Needs to be tested and then either we make an override or maintain code ourselves, as it does not look like upstream agrees with
this approach.

@S-S-X @OgelGames @6r1d

**Comment by BuckarooBanzay** [comment]
> as it does not look like upstream agrees with this approach.

Did you bring that up to cheapie? Are you sure this does not work out somehow? :confused: 

I saw your PR on the mirror repo:
https://github.com/minetest-mirrors/digistuff/pull/1

This is one of those brilliant free software situations where a new fork is created, is it? :smirk: 
How can we still get future maintenance- and feature-updates?

At first glance it looks like this can be done with overrides too (the description and node-definition fields) wouldn't that be a viable option too?

Anyway, i'm neutral to this and can live with either decision (i do have a few enhancements though if we would fork it...)

**Comment by SwissalpS** [comment]
Yes, I think we can easily fix situation with an override. Take the list from my fork and process it in a pandora_custom script.

**Comment by S-S-X** [comment]
> i do have a few enhancements though if we would fork it...

Implementing #183 would also become trivial task, also cleanup of [stuff added](https://github.com/pandorabox-io/pandorabox_custom/blob/3916c3bff81c4befc08548e28de9f5a28b7d60d1/hacks.lua#L25-L39) for #539

**Comment by OgelGames** [comment]
IMO forking would be better because there are other things to fix, like what @S-S-X mentioned, and also the game controller, which currently doesn't work because of this: https://github.com/minetest-mirrors/digistuff/blob/c4b40bcfd5c7116a48af87e3db07a9c29757fed9/controller.lua#L96

**Comment by SwissalpS** [comment]
[for now, I propose this fix](https://github.com/pandorabox-io/pandorabox_custom/pull/23)
Tests will show if I got some entries wrong. Also they will show which nodes need to
have an update hook added.

**Comment by SwissalpS** [comment]
have tested and updated.
Only force field and supply converter don't work properly yet.
Fancy vendor has it's own tool, so I didn't bother with that.

**Comment by SwissalpS** [comment]
Thanks @S-S-X your link saved me some lookups and now force field and supply converter also work.
